packages/rainbowkit/src/wallets/getWalletConnectConnector.ts

```diff
@@ -1,105 +1,66 @@
-import { WalletConnectConnector } from 'wagmi/connectors/walletConnect';
-import { WalletConnectLegacyConnector } from 'wagmi/connectors/walletConnectLegacy';
-import { Chain } from '../components/RainbowKitProvider/RainbowKitChainContext';
+import { createConnector } from 'wagmi';
+import { CreateConnectorFn } from 'wagmi';
+import { walletConnect } from 'wagmi/connectors';
+import { CreateConnector, WalletOptionsParams } from './Wallet';
 
-type SerializedOptions = string;
-const sharedConnectors = new Map<
-  SerializedOptions,
-  WalletConnectLegacyConnector | WalletConnectConnector
+const walletConnectInstances = new Map<
+  string,
+  ReturnType<typeof walletConnect>
 >();
 
-type WalletConnectVersion = '1' | '2';
+// Function to get or create a walletConnect instance
+const getOrCreateWalletConnectInstance = (
+  projectId: string,
+  showQrModal: boolean,
+): ReturnType<typeof walletConnect> => {
+  const key = `${projectId}_${showQrModal}`;
 
-type WalletConnectConnectorConfig = ConstructorParameters<
-  typeof WalletConnectConnector
->[0];
-export type WalletConnectConnectorOptions =
-  // @ts-ignore
-  WalletConnectConnectorConfig['options'];
+  const sharedWalletConnector = walletConnectInstances.get(key);
```

Comment:
This function will do very similar to what we do [here](https://github.com/rainbow-me/rainbowkit/blob/bd694e8457260cd3dd1c28a2076866d2a61f1d11/packages/rainbowkit/src/wallets/getWalletConnectConnector.ts#L102) just instead we will store it as a function so it can be reusable